rlwe/scale.go

```diff
@@ -13,16 +13,30 @@ const (
 
 // Scale is a struct used to track the scaling factor
 // of Plaintext and Ciphertext structs.
-// The scale is managed as an 128-bit precision real.
+// The scale is managed as an 128-bit precision real and can
+// be either a floating point value or a mod T
+// prime integer, which is determined at instantiation.
 type Scale struct {
 	Value big.Float
+	Mod   *big.Int
 }
 
-// NewScale instantiates a new Scale.
-// Accepted types are int, int64, uint64, float64, *big.Int, *big.Float and *Scale.
+// NewScale instantiates a new floating point Scale.
+// Accepted types for s are int, int64, uint64, float64, *big.Int, *big.Float and *Scale.
 // If the input type is not an accepted type, returns an error.
-func NewScale(s1 interface{}) (s2 Scale) {
-	return Scale{Value: *scaleToBigFloat(s1)}
+func NewScale(s interface{}) Scale {
+	return Scale{Value: *scaleToBigFloat(s)}
+}
+
+// NewScaleModT instantiates a new integer mod T Scale.
+// Accepted types for s are int, int64, uint64, float64, *big.Int, *big.Float and *Scale.
+// If the input type is not an accepted type, returns an error.
+func NewScaleModT(s interface{}, mod uint64) Scale {
```

Comment:
This is actually not going to work if we need to multiply the scale of two different schemes in a generic way. I either need an interface-based scale creation, from the params, using the default scale stored in the `rlwe.Parameters` or an empty interface for the method on the `rlwe.Scale`, like `a.Mul(b interface{})`.

Reply:
I added a generic way to create the scale from the `rwle.Parameters`